records/generator_test.go

```diff
@@ -267,3 +273,26 @@ func TestHashString(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestTimeout(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Integration test - skipping for short mode.")
+	}
+
+	sleepForeverHandler := func(w http.ResponseWriter, req *http.Request) {
+		req.Close = true
+		notify := w.(http.CloseNotifier).CloseNotify()
+		<-notify
+	}
+	server := httptest.NewServer(http.HandlerFunc(sleepForeverHandler))
+	defer server.Close()
+
+	rg := NewRecordGenerator(500 * time.Millisecond)
+	host, port, err := net.SplitHostPort(server.Listener.Addr().String())
+	_, err = rg.loadFromMaster(host, port)
+	netErr := (err).(*url.Error).Err.(net.Error)
```

Comment:
if might make for a more readable failure if there some additional assertions here:
- `if err == nil { f.Fatalf("expected error because ...") }`
- `urlErr, ok := ...; if !ok { t.Fatalf("expected url.Error instead of %#+v", err) }`

... etc